sql/snowflake/SnowflakeLexer.g4

```diff
@@ -835,6 +835,7 @@ SNAPPY:                                                'SNAPPY';
 SNAPPY_COMPRESSION:                                    'SNAPPY_COMPRESSION';
 SNOWFLAKE_FULL:                                        'SNOWFLAKE_FULL';
 SNOWFLAKE_SSE:                                         'SNOWFLAKE_SSE';
+SNOWPARK_OPTIMIZED:                                    '\'SNOWPARK-OPTIMIZED\'';
```

Comment:
It looks like it should be added to the [`character`](https://github.com/antlr/grammars-v4/blob/master/sql/snowflake/SnowflakeParser.g4#L2014) rule since I guess it's a valid literal as well as [`CHAR_LITERAL`](https://github.com/antlr/grammars-v4/blob/master/sql/snowflake/SnowflakeLexer.g4#L1130).

Please also add other tokens enclosed with quotes (`AAD_PROVISIONER_Q`, `AUTO_Q`, and so on). Or at least raise an issue on this problem.